smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -339,17 +379,70 @@ private void writeShapeValidator(TypeScriptWriter writer, Shape shape, Collectio
             } else {
                 writer.writeInline("new __CompositeMapValidator<$T>", symbolProvider.toSymbol(valueShape));
             }
-            writer.openBlock("(", "),", () -> {
-                writeCompositeValidator(writer, mapShape, constraintTraits);
+            writer.openBlock("(", ")" + trailer, () -> {
+                writeCompositeValidator(writer, mapShape, constraintTraits, ",");
                 writeShapeValidator(writer,
                                     model.expectShape(keyShape.getTarget()),
-                                    getConstraintTraits(keyShape));
+                                    getConstraintTraits(keyShape),
+                                    ",");
                 writeShapeValidator(writer,
                                     model.expectShape(valueShape.getTarget()),
-                                    getConstraintTraits(valueShape));
+                                    getConstraintTraits(valueShape),
+                                    "");
             });
         } else if (shape instanceof SimpleShape) {
-            writeCompositeValidator(writer, shape, constraintTraits);
+            writeCompositeValidator(writer, shape, constraintTraits, trailer);
+        } else {
+            throw new IllegalArgumentException(
+                    String.format("Unsupported shape found when generating validator: %s", shape));
+        }
+    }
+
+    private void writeShapeValidatorType(TypeScriptWriter writer, Shape shape) {
```

Comment:
I am going to evaluate if I can remove some duplication between this and the validator-writing method by using `writeInline`.

Reply:
I did it, but arguably it made things uglier. More compact, though.